### PR TITLE
PLT-4752 Add editOAuthApp action

### PR DESCRIPTION
**Pull request by n1aba**
#### Summary
Add editOAuthApp action, test and missing initial state for OAuth requests

#### Ticket Link
https://github.com/mattermost/platform/issues/4752

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

#### Test Information
This PR was tested on: [Desktop Ubuntu 16.04 LTS Google Chrome 58.0.3029.110 (64-bit)]

### CODE REVIEW

File: src/client/client4.js [hunk 1]
```diff
@@ -1584,6 +1584,13 @@ export default class Client4 {
         );
     };
 
+    editOAuthApp = async (app) => {
+        return this.doFetch(
+            `${this.getOAuthAppsRoute()}/${app.id}`,
```
Comment (enahum) [inline src/client/client4.js]: looks like this route does not exists? thus the build failure

```
1) Actions.Integrations editOAuthApp:
     Error: {"message":"Sorry, we could not find the page.","server_error_id":"api.context.404.app_error","status_code":404,"url":"http://localhost:40914/api/v4/oauth/apps/qtodcceyfj8hden61g6ifiq6jc"}
      at _callee22$ (test/actions/integrations.test.js:499:19)
      at tryCatch (node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)
      at Generator.invoke [as _invoke] (node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)
      at Generator.prototype.(anonymous function) [as next] (node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)
      at step (test/actions/integrations.test.js:33:191)
      at test/actions/integrations.test.js:33:361
      at process._tickCallback (internal/process/next_tick.js:103:7)
```

Comment (n1aba) [reply src/client/client4.js]: It does not exist yet but I have PR for that https://github.com/mattermost/platform/pull/7361
I'll rerun build once it gets merged

**Comment by jwilander** [comment]
@n1aba did the server PR get merged?

**Comment by n1aba** [comment]
Nope, it says "This commit cannot be built" and I have no idea why.
https://github.com/mattermost/mattermost-server/pull/7413

**Comment by n1aba** [comment]
Also The Details link for "continuous-integration/jenkins/pr-merge" requires credentials

**Comment by jwilander** [comment]
Oh, sorry that's from our new build server that isn't quite stable yet. I've marked that ready to merge and just waiting on the build.mattermost.com build to pass. Apologies for it taking this long, your PR seemed to have slipped through the cracks

**Comment by n1aba** [comment]
Does jenkins use mattermost-server under master branch here? Tests are passing locally with latest server code

**Comment by n1aba** [comment]
I've mocked failing test

**Comment by cpanato** [comment]
this PR https://github.com/mattermost/mattermost-server/pull/7413 got merged. waiting this one to review the https://github.com/mattermost/mattermost-webapp/pull/44 after PM review

**Pull request merged**